gno.land/pkg/gnoclient/client_txs.go

```diff
@@ -8,77 +8,86 @@ import (
 	"github.com/gnolang/gno/tm2/pkg/std"
 )
 
+// MsgCall - syntax sugar for vm.MsgCall
+type MsgCall struct {
+	PkgPath  string   // Package path
+	FuncName string   // Function name
+	Args     []string // Function arguments
+	Send     string   // Send amount
+}
+
 // CallCfg contains configuration options for executing a contract call.
 type CallCfg struct {
-	PkgPath        string   // Package path
-	FuncName       string   // Function name
-	Args           []string // Function arguments
-	GasFee         string   // Gas fee
-	GasWanted      int64    // Gas wanted
-	Send           string   // Send amount
-	AccountNumber  uint64   // Account number
-	SequenceNumber uint64   // Sequence number
-	Memo           string   // Memo
+	Msgs           []MsgCall
+	GasFee         string // Gas fee
+	GasWanted      int64  // Gas wanted
+	AccountNumber  uint64 // Account number
+	SequenceNumber uint64 // Sequence number
+	Memo           string // Memo
 }
 
 // Call executes a contract call on the blockchain.
 func (c *Client) Call(cfg CallCfg) (*ctypes.ResultBroadcastTxCommit, error) {
 	// Validate required client fields.
 	if err := c.validateSigner(); err != nil {
-		return nil, errors.Wrap(err, "validate signer")
+		return nil, err
 	}
 	if err := c.validateRPCClient(); err != nil {
-		return nil, errors.Wrap(err, "validate RPC client")
+		return nil, err
 	}
 
-	pkgPath := cfg.PkgPath
-	funcName := cfg.FuncName
-	args := cfg.Args
-	gasWanted := cfg.GasWanted
-	gasFee := cfg.GasFee
-	send := cfg.Send
-	sequenceNumber := cfg.SequenceNumber
-	accountNumber := cfg.AccountNumber
-	memo := cfg.Memo
-
-	// Validate config.
-	if pkgPath == "" {
-		return nil, errors.New("missing PkgPath")
-	}
-	if funcName == "" {
-		return nil, errors.New("missing FuncName")
+	msgs := make([]vm.MsgCall, 0, len(cfg.Msgs))
+	for _, msg := range cfg.Msgs {
+		// Validate config.
+		if msg.PkgPath == "" {
+			return nil, errInvalidPkgPath
+		}
+		if msg.FuncName == "" {
+			return nil, errInvalidFuncName
+		}
+
+		// Parse send amount.
+		sendCoins, err := std.ParseCoins(msg.Send)
+		if err != nil {
+			return nil, errors.Wrap(err, "parsing send coins")
+		}
+
+		if err != nil {
+			return nil, errors.Wrap(err, "parsing gas fee coin")
+		}
+
+		caller := c.Signer.Info().GetAddress()
```

Comment:
This can be inlined

Reply:
[ced737b](https://github.com/gnolang/gno/pull/1565/commits/ced737b4d07ebf1521ae5929c0f3b337d258a841)